### DISCUSSION TITLE
Deferred links not working on iOS

### DISCUSSION THREAD

**Discussion by barbhackk** [invertase/react-native-firebase#6372] [Q&A]
Hello,

I'm having a problem with differed deep links only on iOS (on Android it works fine). Before it was working fine. I feel like since updating to iOS 15.5 it doesn't work anymore (maybe even since iOS 14).

I've scoured gitHub for a similar issue but can't find anything. I did follow the react-native-firebase and dynamicLink installation guide but it still doesn't work.

```
"dependencies": {
    "@react-native-async-storage/async-storage": "~1.15.0",
    "@react-native-firebase/app": "^15.1.1",
    "@react-native-firebase/dynamic-links": "^15.1.1",
```

I'm on iOS 15.5

```
"react-native": "0.64.3"
```

After clicking on the deferred link, it sends me back to the app store. I install the app and open it.

```js
// Dynamic Link - Deferred Deep Link
dynamicLinks()
            .getInitialLink()
            .then(link => {
                if (link) {
                    setTimeout(() => {
                        this._navigate(link.url);
                    }, 500);
                }
            })
```

link is 'null' every time.

Thanks for help,

Sebastien

**Comment by mikehardy** [comment]
You might check the firebase-ios-sdk repo to see if they've got any relevant issues in flight :thinking:

**Reply by barbhackk** [reply]
I looked around and couldn't find anything about it. Before it worked and since the last iOS update nothing. I didn't touch this part. If I knew how to check that a deferred deep link is in memory somewhere!

**Reply by barbhackk** [reply]
Am I missing the firebase.json file? Nope ?

**Comment by barbhackk** [comment]
Hi,

I found something. The dynamic link clicked ends with « &efr=1 ». It avoids me having the preview page after clicking and thus redirecting directly to the AppStore.

Except that with this url, the link is not copied to the clipboard and dynamicLink finds a null value.

If I put &efr=0 it works well. 😳

https://firebase.google.com/docs/dynamic-links/create-manually?hl=en

**Reply by barbhackk** [reply]
See this comment https://github.com/firebase/firebase-ios-sdk/issues/3719

**Reply by mikehardy** [reply]
@barbhackk that is a really informative link and has already helped an unrelated person, thank you for posting it! I don't think this is something under react-native-firebase control though apparently - this is in the firebase-ios-sdk internal workings
